site/content/contribute/webapp/migrating-to-typescript.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "Migrating to Typescript"
+date: 2020-07-11T23:00:00-04:00
+weight: 9
+subsection: Web App
+---
+
+The Mattermost team wants to improve the quality, security and stability of the code, and one way to do this is by introducing the usage of type checking. Thus, we have decided to introduce Typescript in our codebase. As a first step, we have migrated the [mattermost-redux](https://github.com/mattermost/mattermost-redux) library to use Typescript, and are now in the processs of migrating the [webapp](https://github.com/mattermost/mattermost-webapp) to use Typescript.
+
+This campaign will help with the migration by converting files written in Javascript to type-safe files written in Typescript.
+
+By completing this campaign, we are looking to:
+
+- reduce the errors derived from changes.
+- increase the consistency of the code.
+- ensure a more defensive programming in the code.
```

Comment:
```suggestion
- Ensure a more defensive programming in the code.
```

Reply:
Comments:
Do we have stats around current consistency and errors, and whether this change will significantly improve them?

For point three, I wonder if it would also work if we said:
"Create a more defensive strategy for writing the code." > I'm not 100% certain what the point refers to, so that may be way off base. :)

Reply:
> Do we have stats around current consistency and errors, and whether this change will significantly improve them?

I don't think so. It seems like a pretty difficult thing to track. One objective benefit you could argue is that having a strongly-typed language helps cover all branches of your code, rather than just what unit tests are written to cover.

Reply:
That's a good benefit - should we include that in the opening paragraph?